src/OrchardCore/Orchard.Environment.Shell/FileShellSettingsConfigurationProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+
+namespace Orchard.Environment.Shell
+{
+    public class FileShellSettingsConfigurationProvider : IShellSettingsConfigurationProvider
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+
+        public FileShellSettingsConfigurationProvider(IHostingEnvironment hostingEnvironment)
+        {
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        public int Order => 1000;
+
+        public void AddSource(IConfigurationBuilder builder)
+        {
+            builder.AddJsonFile(Path.Combine(_hostingEnvironment.ContentRootPath, "tenants.json"));
+        }
+
+        public void SaveToSource(string name, IDictionary<string, string> configuration)
```

Comment:
Yup - that's the point of this. The idea being I can have some config values stored locally, and others in AWS KMS, I pull both sources in, and let the MS configuration libs do the stitching for me.